### PR TITLE
Chris taco

### PR DESCRIPTION
**Pull request by chris-f-anderson**

### CODE REVIEW

File: LoggingKata.Test/TacoParserTests.cs [hunk 1]
```diff
@@ -12,18 +12,41 @@ public void ShouldDoSomething()
...
+            //Arrange
+            TacoParser tacoparser = new TacoParser();
+
+            //Act
+            ITrackable actual = tacoparser.Parse(str);
+
+            //Assert
+            Assert.NotNull(actual);
+            
             // TODO: Complete Should Parse
```
Comment (chipelmer) [inline LoggingKata.Test/TacoParserTests.cs:29]: This is done now, so the "TODO" should be removed

File: LoggingKata.Test/TacoParserTests.cs [hunk 2]
```diff
@@ -12,18 +12,41 @@ public void ShouldDoSomething()
...
-        [InlineData("")]
-        public void ShouldFailParse(string str)
+        [InlineData(null, null)]
+        [InlineData(" ", null)]
+        [InlineData("Me", null)]
+
+
+        public void ShouldFailParse(string str, ITrackable expected)
         {
             // TODO: Complete Should Fail Parse
```
Comment (chipelmer) [inline LoggingKata.Test/TacoParserTests.cs:40]: This is done now, so the "TODO" should be removed

File: LoggingKata.Test/TacoParserTests.cs [hunk 3]
```diff
@@ -12,18 +12,41 @@ public void ShouldDoSomething()
...
+
+            //Arrange
+            TacoParser tacoParser = new TacoParser();
+
+            //Act
+            ITrackable actual = tacoParser.Parse(str);
+
+            //Assert
+            Assert.Equal(actual, expected);
```
Comment (chipelmer) [inline LoggingKata.Test/TacoParserTests.cs:49]: Since all the failed parses should return null, this assertion could instead be `Assert.Null(actual);`

File: LoggingKata/Program.cs [hunk 4]
```diff
@@ -21,6 +21,32 @@ static void Main(string[] args)
...
             var locations = lines.Select(parser.Parse).ToArray();
+            ITrackable locA = null;
+            ITrackable locB = null;
+            Double maxDistance = 0;
+
+            for (int i = 0; i < locations.Length; i++)
+            {
+                GeoCoordinate corA = new GeoCoordinate();
+                corA.Latitude = locations[i].Location.Latitude;
+                corA.Longitude = locations[i].Location.Longitude;
```
Comment (chipelmer) [inline LoggingKata/Program.cs:32]: The GeoCoordinate class has a constructor overload that receives a latitude and longitude as parameters, so these 3 lines could be combined as `GeoCoordinate corA = new GeoCoordinate(locations[i].Location.Latitude, locations[i].Location.Longitude);`
This is a pretty long line, though, so it may not be preferred.

File: LoggingKata/Program.cs [hunk 5]
```diff
@@ -21,6 +21,32 @@ static void Main(string[] args)
...
+                for (int k = 0; k < locations.Length; k++)
+                {
+                    GeoCoordinate corB = new GeoCoordinate();
+                    corB.Latitude = locations[k].Location.Latitude;
+                    corB.Longitude = locations[k].Location.Longitude;
+
+                    if (corA.GetDistanceTo(corB) >= maxDistance)
+                    {
+                        maxDistance = corA.GetDistanceTo(corB);
```
Comment (chipelmer) [inline LoggingKata/Program.cs:41]: Instead of calculating this twice, it's probably better to calculate it once and then use the saved value for the if-statement and here.  Something like:
```CS
double currentDistance = corA.GetDistanceTo(corB);
if (currentDistance >= furthestDistance)
{
  furthestDistance = currentDistance;
  // ...
```

File: LoggingKata/TacoParser.cs [hunk 6]
```diff
@@ -8,11 +8,34 @@ public class TacoParser
...
+            }
+
+            var cells = line.Split(',');
+
+            // If your array.Length is less than 3, something went wrong
+            if (cells.Length < 3)
+            {
+                // Log that and return null
+                logger.LogError("Error");
+                return null; // TODO Implement
```
Comment (chipelmer) [inline LoggingKata/TacoParser.cs:26]: This is done now, so the "TODO" should be removed

File: LoggingKata/TacoParser.cs [hunk 7]
```diff
@@ -8,11 +8,34 @@ public class TacoParser
...
+            }
+
+            double lattitude, longitude;
+            double.TryParse(cells[0], out lattitude);
+            double.TryParse(cells[1], out longitude);
```
Comment (chipelmer) [inline LoggingKata/TacoParser.cs:31]: The success/failure of these TryParse methods isn't used.  There should be an if-statement to handle whether or not they worked:
```CS
bool latitudeSuccess = double.TryParse(cells[0], out lattitude);
bool longitudeSuccess = double.TryParse(cells[1], out longitude);
if (latitudeSuccess == false || longitudeSuccess == false)
{
  return null;
}
// ...
```

File: LoggingKata.Test/TacoParserTests.cs [hunk 8]
```diff
@@ -12,18 +12,41 @@ public void ShouldDoSomething()
...
             // TODO: Complete Should Parse
         }
 
         [Theory]
-        [InlineData(null)]
-        [InlineData("")]
-        public void ShouldFailParse(string str)
+        [InlineData(null, null)]
+        [InlineData(" ", null)]
+        [InlineData("Me", null)]
```
Comment (chipelmer) [inline LoggingKata.Test/TacoParserTests.cs:35]: There are more things that could be passed in as bad input data, especially around edge cases.  Things like `"10, ten, Taco Bell"` that make sure that the details of parsing the string are working properly

**Review by chipelmer — APPROVED** [review]
This looks good, well done 👍
